### PR TITLE
Change SocketCluster listeners from raw to dedicated event - Closes #26

### LINKED ISSUE #26: Change SocketCluster listeners from raw to dedicated event
As explained in https://github.com/LiskHQ/lisk/issues/1472, client WS connections should be using `socket.emit()` instead of `socket.send()` while executing `RPC` calls. To handle that change, `WAMPServer.js` needs to accept for a dedicated event instead of the`raw` one.

### PR DESCRIPTION
**Pull request by MaciejBaj**
### What was the problem?
- `wamp-socket-cluster` was using low-level SocketCluster `send` call and `raw` listeners
- `subscibe/publish`  part of WAMP protocol might be handled by `SocketCluster` itself

### How did I fix it?
- get rid of handling events by `wamp-socket-cluster`. Pass them directly to SocketCluster
- introduce dedicated `rpc-request` and `rpc-response` event endpoints and access them using `emit` function
- get rid of misleading `WAMP` terminology from schemas, as it handles `RPC` only
- upgrade prepared example, so the emit and rpc example functions are: returning a valid result, error or not replying at all with equal probability

### How to test it?
Run example

Closes #26

### CODE REVIEW

**Review by jondubois — COMMENTED** [review]
Based on our discussion, we could simplify the logic even further by using a response handler function to handle the RPC response:

E.g.
```
socket.emit('rpc-request', rpcDetails, responseHandlerFunction)
```

If we use this approach:
- We won't need to keep track of the response handlers and deal with timeout errors ourselves and;
- The RPC response handlers will be able to detect a lost connection sooner.

File: SlaveWAMPServer.js [hunk 1]
```diff
@@ -25,14 +25,14 @@ class SlaveWAMPServer extends WAMPServer {
 		this.endpoints.slaveRpc = {};
 		this.config = {};
 		this.internalRequestsTimeoutMs = internalRequestsTimeoutMs;
+		this.configuredCb = configuredCb;
```
Comment (jondubois) [inline SlaveWAMPServer.js]: Just a note: If `configuredCb` is not used outside of the constructor, maybe we don't need to assign it to `this`?

**Review by jondubois — DISMISSED** [review]
Looks much better now. I just found one minor thing but don't worry about it.

**Review by karmacoma — DISMISSED** [review]

File: SlaveWAMPServer.js (hunk 1 above)
Comment (MaciejBaj) [reply SlaveWAMPServer.js]: True, thanks!

**Review by jondubois — APPROVED** [review]
:+1:

**Review by karmacoma — APPROVED** [review]

**Pull request merged**